### PR TITLE
Migrate to Github Actions

### PR DESCRIPTION
**Pull request by vatanaksoytezer**
Realized this branch is missing GHA and travis is not being build right now. This PR is to migrate this branch to GHA.

### CODE REVIEW

File: .ci.prepare_codecov [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
```
Comment (henningkayser) [inline .ci.prepare_codecov:1]: CI fails because of this, does this file need x permissions or so?

**Review by henningkayser — COMMENTED** [review]
Looks good, I think we only need to fix codecov

File: .ci.prepare_codecov (hunk 1 above)
Comment (vatanaksoytezer) [reply .ci.prepare_codecov:1]: Oh maybe that, let me try.

**Review by henningkayser — APPROVED** [review]
Well, that worked :+1:

**Pull request merged**